engine/table/src/main/java/io/deephaven/engine/table/impl/by/StaticSymbolTableChunkedOperatorAggregationStateManager.java

```diff
@@ -0,0 +1,136 @@
+        Arrays.fill(keyPositions, -1);
+    }
+
+    @Override
+    public int maxTableSize() {
+        return MAX_ARRAY_SIZE;
+    }
+
+    @Override
+    public SafeCloseable makeAggregationStateBuildContext(final ColumnSource<?>[] buildSources, long maxSize) {
+        return null;
+    }
+
+    @Override
+    public void add(final SafeCloseable bc, final RowSequence orderedKeys, final ColumnSource<?>[] sources, final MutableInt nextOutputPosition, final WritableIntChunk<RowKeys> outputPositions) {
+        if (orderedKeys.isEmpty()) {
+            return;
+        }
+
+        outputPositions.setSize(orderedKeys.intSize());
+
+        try (final RowSequence.Iterator okIt = orderedKeys.getRowSequenceIterator();
+             final ChunkSource.FillContext fillContext = mappedKeySource.makeFillContext(CHUNK_SIZE);
+             final WritableLongChunk<RowKeys> symbolTableValues = WritableLongChunk.makeWritableChunk(tableSize);
+             final WritableIntChunk<Values> symbolLookupChunk = WritableIntChunk.makeWritableChunk(CHUNK_SIZE) ) {
+
+            symbolTableValues.setSize(0);
+
+            final int firstNewPosition = nextPosition;
+
+            while (okIt.hasMore()) {
+                final RowSequence nextKeys = okIt.getNextRowSequenceWithLength(CHUNK_SIZE);
+                final LongChunk<Values> symbolSourceChunk = mappedKeySource.fillChunkWithSymbolSource(fillContext, symbolLookupChunk, nextKeys);
+
+                final int chunkSize = symbolLookupChunk.size();
+                for (int ii = 0; ii < chunkSize; ii++) {
+                    final int key = symbolLookupChunk.get(ii);
+                    if (key == QueryConstants.NULL_INT) {
+                        if (nullPosition == -1) {
+                            nullPosition = nextPosition++;
+                        }
+                        outputPositions.set(ii, nullPosition);
+                    } else {
+                        final int keyPosition = keyPositions[key];
+                        if (keyPosition == -1) {
+                            outputPositions.set(ii, keyPositions[key] = nextPosition++);
+                            symbolTableValues.add(symbolSourceChunk.get(ii));
+                        } else {
+                            outputPositions.set(ii, keyPosition);
+                        }
+                    }
+                }
+            }
+
+            if (nextPosition != firstNewPosition) {
+                updateKeyHashTableSources(symbolTableValues, firstNewPosition);
+            }
+        }
+
+        nextOutputPosition.setValue(nextPosition);
+    }
+
+    private void updateKeyHashTableSources(final WritableLongChunk<RowKeys> symbolTableValues, final int firstNewPosition) {
+        keyColumn.ensureCapacity(nextPosition);
+
+        final ColumnSource<?> symbolColumnSource = symbolTable.getColumnSource(SymbolTableSource.SYMBOL_COLUMN_NAME);
+
+        int symbolIterIdx = 0;
+        int nextPosition = firstNewPosition;
```

Comment:
Please do something about this name. It's less clear than it could be by hiding the instance member.